app/controllers/new-signup.js

```diff
@@ -59,12 +69,56 @@ export default class NewSignUpController extends Controller {
 
   @action handleInputChange(key, value) {
     this.error = '';
-    set(this.signupDetails, key, value);
+    if (key === 'role') {
+      if (value === 'developer') {
+        this.developer = true;
+        this.designer = false;
+        this.mavens = false;
+        this.productManger = false;
+      } else if (value === 'designer') {
+        this.developer = false;
+        this.designer = true;
+        this.mavens = false;
+        this.productManger = false;
+      } else if (value === 'mavens') {
+        this.developer = false;
+        this.designer = false;
+        this.mavens = true;
+        this.productManger = false;
+      } else if (value === 'productManager') {
+        this.developer = false;
+        this.designer = false;
+        this.mavens = false;
+        this.productManger = true;
+      }
+    } else {
+      set(this.signupDetails, key, value);
+    }
     if (this.signupDetails[key] > '') this.isButtonDisabled = false;
-    else this.isButtonDisabled = true;
+    else if (
+      this.developer > '' ||
+      this.designer > '' ||
+      this.mavens > '' ||
+      this.productManger > ''
+    ) {
+      this.isButtonDisabled = false;
+    } else this.isButtonDisabled = true;
   }
 
   @action async signup() {
+    let roles;
+
+    if (this.dev) {
+      if (this.developer) {
+        roles = { developer: this.developer };
+      } else if (this.designer) {
+        roles = { designer: this.designer };
+      } else if (this.mavens) {
+        roles = { mavens: this.mavens };
+      } else {
+        roles = { productManager: this.productManger };
+      }
```

Comment:
Once the object for roles is created, the selected option should be directly available here, which can be used instead of if-else condition

Reply:
Done with changes can you please check my latest commit